### DISCUSSION TITLE
Auto Fan Curve - Step Size

### DISCUSSION THREAD

**Discussion by Sniper435** [Rem0o/FanControl.Releases#691] [Q&A]
Can anyone shed any light on the auto fan curve and the step size it uses when setting the speed %?

I've been playing with the auto curves to see how well they work to control my CPU and GPU temperatures and am pretty happy with them, the reason I ask about step size is that I notice the CPU curve regularly moving between two % values (30% and 47.5%) without ever using intermediate speeds.

Is this expected? is it caused by something in my settings? could the auto curve use smaller steps in % to give finer control?

![image](https://user-images.githubusercontent.com/6918097/165507524-05e637c4-5d54-4ad7-bf44-b2a70ef8c7e9.png)
For reference, my system does idle under 40C when the fans are at 35% so the settings I've got should allow for that, rather than the fans oscillating between 30% and 47.5%

**Comment by Rem0o** [comment]
It will bump the base speed when it leaves the idle state.  The "auto" part is really meant around the "load" state, where the temp is close the the load temperature.  I think you should increase your idle temp to like 45, or maybe 50, to increase that threshold a bit and reflect your actual system better.

**Comment by Sniper435** [comment]
OK - so you'd expect the idle temp to be set to the lower end of the expected load temperatures?
is there a level of granularity used in the fan % calculations that would explain the large steps I'm seeing?

I'll give your suggestion a go and give further feedback

**Reply by Rem0o** [reply]
In any case, don't forget the set your step% on your controls.

**Reply by Sniper435** [reply]
these are my current settings for reference:
![image](https://user-images.githubusercontent.com/6918097/165546828-5a69dce8-e7c7-4694-98f2-226dc8b9df59.png)

setting the idle temp higher on the Auto CPU curve does mean I don't notice the fans spinning up and down so much as they tend to sit at 30% most of the time, when they do change though it is still with a large jump of 17.5% to 47.5%. this happens as soon as the temperature is over the idle temp of 50C and as soon as it dips below the fans go back to 30%.
I was expecting that there would be more gradual changes in fan speed, allowing the system to find a more stable state rather than needing to step the fans up and down in such large jumps
